src/pages/ExamsPage/components/ExamSelection.test.jsx

```diff
@@ -44,8 +44,8 @@ describe('ExamSelection', () => {
     const input = screen.getByPlaceholderText('Search for an exam...');
     await user.type(input, 'exam1');
     expect(screen.getByText('exam1')).toBeInTheDocument();
-    expect(screen.queryByText('exam2')).not.toBeInTheDocument();
-    expect(screen.queryByText('exam3')).not.toBeInTheDocument();
+    expect(screen.queryByText('exam2')).toBeInTheDocument();
```

Comment:
Updated these tests since these exams should now be visible, though disabled. That being said, I'm not sure how to best test that exam2 and exam3 are disabled. I was going to try to test that the prop was set correctly, but after doing some research, this seems to go against rtl best practices/testing framework. Maybe I could test that if I try to click on exam 2 or 3, that nothing will happen? Any thoughts/ideas here are appreciated. Thanks!

Reply:
Ah I should have clarified, I did try that and that didn't work. I'm wondering if disabled means something different for the MenuItem component vs for the element. But maybe I didn't use it correctly, so I'll try that again/make sure I'm calling the right thing.

Reply:
Ah, darn!

I took a look at the implementation of `MenuItem`, and it looks like it will just pass the `disabled` prop on through to the default `button` element (since you don't specify an `as` prop).

Based on the documentation for `toBeDisabled`, I think your intuition that you might be asserting the matcher on the "wrong" thing could be right. How have you queried for the element you're using `toBeDisabled` on? What kind of elements are you getting back?

Reply:
So I tried a slightly different approach. Let me know what you think! The test now searches for exam1 (as it did before), then clicks on exam2 (which should be disabled). Since it's disabled, the mockHandleSelectExam function should not be called. So this test now just checks that it is not called for the disabled search text. Does this feel like it covers the change in implementation appropriately?

Reply:
I think that assertion definitely covers it, but I do think it would be better to make use of the `toBeDisabled` matcher, because it's more semantically correct than checking whether the click handler runs. There's a definition of what it means to be disabled, and I think it would be better to leverage that. For example, if the click handler ever changes on that button, this test would still pass, because the click handler won't be called when clicking the button. That tells me that this assertion isn't exactly testing whether the button is disabled. By checking whether the button is disabled, we don't need to worry about the value of the click handler changing.

I think something like this should work. Have you tried something like this?

```
const button = screen.queryByRole('link', { name: 'exam2' });
expect(button).toBeDisabled();
```

I know you mentioned that `toBeDisabled` didn't work for you. Did you have something like this before? I'm assuming you may have because I see calls to `getByText` in your tests.

```
expect(screen.getByText('exam2')).toBeDisabled();
```

The reason that this won't work is because the `screen.getByText('exam2')` call is returning the `span` element that wraps the "exam2" text, not the actual `button` element. The `span` element isn't the one that's disabled. The `button` element is the one that's disabled, because it gets the `disabled` prop.

That's why I've used the `queryByRole` function instead, because it'll return to us the actual `link` element (it's a `button` element but has the "role=link" prop, so RTL treats it like it has the `link` role).

Reply:
Ooohhhh, this is very clarifying and helpful! Thank you. I see what you're saying about the test not being as semantically correct.

Yes, exactly! I had followed the existing tests and used getByText. It worked with your suggestion! I'll do some reading on these concepts so that I fully understand/retain for the future. Again, thanks for your help on this! :)